.github/workflows/ci.yml

```diff
@@ -40,35 +51,73 @@ jobs:
         run: npm run test:coverage
 
       - name: Upload coverage results to Coveralls
-        uses: coverallsapp/github-action@master
+        uses: coverallsapp/github-action@v2
+        if: ${{ matrix.os == 'ubuntu-latest' && matrix.node == '20' }}
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           path-to-lcov: ./test/coverage/lcov.info
 
+  security:
+    name: Security audit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run security audit
+        run: npm audit --audit-level=moderate
+
+      - name: Run dependency vulnerability check
+        run: npx audit-ci --moderate
+
```

Comment:
`npm audit` is notoriously noisy and has a high false positive rate - I'd rather we consolidate on dependabot & Snyk instead. cc @staaldraad @doublethink